File: Solutions in JAVA/Strings/ConvertToLowerCase.java
public class ConvertToLowerCase {
	private static String toLowerCase(String str) {
		StringBuilder sb = new StringBuilder(str.length()); // Initialize with known size
		for (int i = 0; i < str.length(); i++) {
			sb.append((char)(str.charAt(i) | 32)); // Bitwise OR to convert uppercase to lowercase
		}
		return sb.toString();
	}

    public static void main(String[] args) {
        System.out.println(toLowerCase("HeLLo WoRLD!")); // Output: hello world!
    }
}

File: Solutions in CPP/strings/convert_to_lowerCase.cpp
#include <iostream>
#include <string>

using namespace std;

string toLowerCase(const string& str) {
    string result;
    result.reserve(str.length()); // Pre-allocate space for efficiency
    
    for (char ch : str) {
        result += (ch >= 'A' && ch <= 'Z') ? (ch | 32) : ch; // Convert only uppercase letters
    }

    return result;
}

int main() {
    cout << toLowerCase("HeLLo WoRLD!") << endl; // Output: hello world!
    return 0;
}
